src/multiexp.rs

```diff
@@ -150,6 +150,39 @@ impl DensityTracker {
     pub fn get_total_density(&self) -> usize {
         self.total_density
     }
+
+    /// Extend by concatenating `other`. If `is_input_density` is true, then we are tracking an input density,
+    /// and other may contain a redundant input for the `One` element. Coalesce those as needed and track the result.
+    pub fn extend(&mut self, other: &mut Self, is_input_density: bool) {
```

Comment:
it should be quite easy to add some unit tests for this, please do so

Reply:
"Tomorrow."